File: sources/c13/nim5/ez8/h5_proto/DllpgdLiteSDK.java
package c13.nim5.ez8.h5_proto;

import IlIlllIIlI1.IllIIlIIII1;
import IllIIlIIII1.llllIIIIll1;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/* loaded from: classes.jar:c13/nim5/ez8/h5_proto/DllpgdLiteSDK.class */
public class DllpgdLiteSDK {
    public static final DllpgdLiteSDK INSTANCE = new DllpgdLiteSDK(IllIIlIIII1.f239llllIIIIll1.llllIIIIll1(new byte[]{75, -105, 77, -98, -86, -100, 1, -34, 67, -110, 66, -123}, new byte[]{47, -5, 33, -18, -51, -8, 47, -67}));
    private final HttpGatewayClient httpClient;

    public DllpgdLiteSDK(String str) {
        this.httpClient = new HttpGatewayClient(str);
    }

    public static JSONObject createSampleAtom() throws JSONException {
        JSONObject jSONObject = new JSONObject();
        jSONObject.put(IllIIlIIII1.llllIIIIll1(new byte[]{21, 34, 68, 111, -51, 21, 79, 13}, new byte[]{113, 71, 50, 6, -82, 112, 6, 105}), IllIIlIIII1.llllIIIIll1(new byte[]{-19, -51, 89, 16, 40, -54, 76, -17, -5, -38, 93, 3, 33, -16, 122, -17}, new byte[]{-98, -84, 52, 96, 68, -81, 19, -117}));
        jSONObject.put(IllIIlIIII1.llllIIIIll1(new byte[]{-30, 11, 79, 91, 14, -9, -37}, new byte[]{-108, 110, 61, 40, 103, -104, -75, -17}), 208);
        jSONObject.put(IllIIlIIII1.llllIIIIll1(new byte[]{74, -71, -122, -96, 48, 19, -56, 89, 76, -84, -72, -111, 60, 21}, new byte[]{43, -55, -10, -16, 81, 112, -93, 56}), IllIIlIIII1.llllIIIIll1(new byte[]{-53, -77, 71, 80, 15, 77, 2, 20, -40, -80, 79, 80, 11, 69, 19}, new byte[]{-88, -36, 42, 126, 106, 53, 99, 121}));
        jSONObject.put(IllIIlIIII1.llllIIIIll1(new byte[]{-17, -28, -106, -74, -47, -111, -37, 74, -31, -6}, new byte[]{-114, -108, -26, -32, -76, -29, -88, 35}), IllIIlIIII1.llllIIIIll1(new byte[]{-16, -7, Byte.MAX_VALUE, 27, -29}, new byte[]{-63, -41, 79, 53, -45, 24, 49, -104}));
        jSONObject.put(IllIIlIIII1.llllIIIIll1(new byte[]{40, -39, -119, 19}, new byte[]{79, -72, -64, 119, 68, -69, -18, -25}), IllIIlIIII1.llllIIIIll1(new byte[]{-50, -100, 103, -13, -30, -26, 84, 57, -36, -94, 99, -25}, new byte[]{-67, -3, 10, -125, -114, -125, 11, 94}));
        jSONObject.put(IllIIlIIII1.llllIIIIll1(new byte[]{124, 97, -64, -51, -29, 37, -105, 121, 107}, new byte[]{15, 4, -77, -66, -118, 74, -7, 48}), IllIIlIIII1.llllIIIIll1(new byte[]{-90, -105, -13, 87, 78, 120, -122, 103, -80, -123, -19, 78, 77, 115, -122, 125, -79}, new byte[]{-43, -10, -98, 39, 34, 29, -39, 20}));
        jSONObject.put(IllIIlIIII1.llllIIIIll1(new byte[]{-116, -124, -53, -54, -88, 83, 40, 105, -120, -104}, new byte[]{-19, -12, -69, -119, -64, 50, 70, 7}), IllIIlIIII1.llllIIIIll1(new byte[]{53, -57, 29, -28, -61, 10, -107}, new byte[]{81, -94, 123, -123, -74, 102, -31, 51}));
        jSONObject.put(IllIIlIIII1.llllIIIIll1(new byte[]{19, -108, -33, 43, -67, -100, -82, 13, 14, -126, -4, 12, -86, -86, -87, 14, 42, -107, -9, 45, -74, -118, -81}, new byte[]{122, -25, -104, 78, -45, -7, -36, 108}), false);
        JSONObject jSONObject2 = new JSONObject();
        jSONObject2.put(IllIIlIIII1.llllIIIIll1(new byte[]{1, -111, 74, -85, -43, -57}, new byte[]{109, -2, 41, -54, -71, -94, 38, 8}), IllIIlIIII1.llllIIIIll1(new byte[]{17, -3, -124, -47, -45}, new byte[]{107, -107, -37, -110, -99, 109, 88, 98}));
        jSONObject2.put(IllIIlIIII1.llllIIIIll1(new byte[]{-14, 93, 47, 93, 117, -29, 112, 118}, new byte[]{-122, 52, 66, 56, 15, -116, 30, 19}), IllIIlIIII1.llllIIIIll1(new byte[]{-86, 88, 97, 60, 50, -125, 61, Byte.MAX_VALUE, -123, 76, 96, 60, 116}, new byte[]{-21, 43, 8, 93, 29, -48, 85, 30}));
        jSONObject2.put(IllIIlIIII1.llllIIIIll1(new byte[]{-102, 28, 117, 42, -56, 29, 28, -4, -113, 24}, new byte[]{-22, 116, 26, 68, -83, 80, 115, -104}), IllIIlIIII1.llllIIIIll1(new byte[]{96, 111, 49, -16, 105, 99, 90, -17, 87, 111}, new byte[]{52, 10, 66, -124, 45, 6, 44, -122}));
        jSONObject2.put(IllIIlIIII1.llllIIIIll1(new byte[]{-125, -110, 32, 85, -47, 59, -123, 16, -121, -114, 55, 78, -47, 60}, new byte[]{-30, -4, 68, 39, -66, 82, -31, 70}), IllIIlIIII1.llllIIIIll1(new byte[]{47, -20}, new byte[]{30, -33, 79, 55, 125, 26, -74, 11}));
        jSONObject2.put(IllIIlIIII1.llllIIIIll1(new byte[]{45, -48, -44, 98, -70, -75, -106, -10, 56, -53, -49, 109, -78, -111}, new byte[]{93, -72, -69, 12, -33, -31, -1, -101}), System.currentTimeMillis());
        jSONObject.put(IllIIlIIII1.llllIIIIll1(new byte[]{45, 84, -123, -18, 94, 28, -3, -95, 47, 94}, new byte[]{73, 49, -13, -121, 61, 121, -76, -49}), jSONObject2);
        jSONObject.put(IllIIlIIII1.llllIIIIll1(new byte[]{0, -85, -59, -107, 77, -39, 52, 25, 22, -88, -61}, new byte[]{112, -57, -80, -14, 36, -73, 125, 119}), new JSONArray());
        return jSONObject;
    }

    public static JSONArray createSampleLogs() throws JSONException {
        JSONArray jSONArray = new JSONArray();
        JSONObject jSONObject = new JSONObject();
        llllIIIIll1 lllliiiill1 = IllIIlIIII1.f239llllIIIIll1;
        jSONObject.put(lllliiiill1.llllIIIIll1(new byte[]{-109, -24, 57, 48, -34}, new byte[]{-1, -115, 79, 85, -78, -106, -126, -79}), 1);
        jSONObject.put(lllliiiill1.llllIIIIll1(new byte[]{43, -26, 111, -46, 33, -82, -50}, new byte[]{70, -125, 28, -95, 64, -55, -85, 66}), lllliiiill1.llllIIIIll1(new byte[]{-77, -80, 61, 17, -47, -51, 5, -47, -113, -74, 112, 12, -40, -37, 86, -36, -121, -76}, new byte[]{-32, -47, 80, 97, -67, -88, 37, -67}));
        jSONObject.put(lllliiiill1.llllIIIIll1(new byte[]{66, 85, 3}, new byte[]{54, 52, 100, 9, 69, -3, 65, 66}), lllliiiill1.llllIIIIll1(new byte[]{-25, 85, 83, -57, -93, 99, 19, -73, -41, 92, 108, -13, -113}, new byte[]{-93, 57, 63, -73, -60, 7, 95, -34}));
        jSONObject.put(lllliiiill1.llllIIIIll1(new byte[]{108, 99, -16, 90, -50, -124, 112, -114, 104}, new byte[]{24, 10, -99, 63, -67, -16, 17, -29}), System.currentTimeMillis());
        jSONArray.put(jSONObject);
        return jSONArray;
    }

    public static JSONArray createSampleEvents() throws JSONException {
        JSONArray jSONArray = new JSONArray();
        JSONObject jSONObject = new JSONObject();
        llllIIIIll1 lllliiiill1 = IllIIlIIII1.f239llllIIIIll1;
        jSONObject.put(lllliiiill1.llllIIIIll1(new byte[]{62, 115, -3, -60}, new byte[]{80, 18, -112, -95, -75, -112, -102, -90}), lllliiiill1.llllIIIIll1(new byte[]{-2, 5, 20, 17, 122, -126, -116, -74, -5, 1, 23, 21}, new byte[]{-115, 100, 121, 97, 22, -25, -45, -45}));
        jSONObject.put(lllliiiill1.llllIIIIll1(new byte[]{-27, Byte.MAX_VALUE, 76, -29}, new byte[]{-127, 26, 63, Byte.MIN_VALUE, -49, -45, 75, -46}), lllliiiill1.llllIIIIll1(new byte[]{5, 13, -86, -7, -68, Byte.MAX_VALUE, -40, 126, 32, 9, -87, -3, -16, 126, -99, 104, 53, 30, -82, -7, -92, 115, -105, 117}, new byte[]{86, 108, -57, -119, -48, 26, -8, 27}));
        jSONObject.put(lllliiiill1.llllIIIIll1(new byte[]{-89, -79, 70, -90, 21, 93, -95, -32, -93}, new byte[]{-45, -40, 43, -61, 102, 41, -64, -115}), System.currentTimeMillis());
        jSONArray.put(jSONObject);
        return jSONArray;
    }

    public JSONObject getConfig(JSONObject jSONObject) throws Exception {
        return this.httpClient.getConfig(jSONObject);
    }

    public CommonResponse updateLog(UpdateLogRequest updateLogRequest) throws Exception {
        return this.httpClient.updateLog(updateLogRequest);
    }

    public CommonResponse updateEvent(UpdateEventRequest updateEventRequest) throws Exception {
        return this.httpClient.updateEvent(updateEventRequest);
    }

    public DllpgdLiteSDK(String str, int i, boolean z) {
        this.httpClient = new HttpGatewayClient(str, i, z);
    }
}

File: python_translation/dllpgd_lite_sdk.py
"""
DllpgdLiteSDK Python Translation

Original Java file: sources/c13/nim5/ez8/h5_proto/DllpgdLiteSDK.java
Package: c13.nim5.ez8.h5_proto

This module provides the main SDK class for data logging and analytics.
Includes sample data generation methods for testing and development.
"""

from typing import Dict, List, Any, Optional
import time


# Note: These classes are referenced but not defined in the provided files
# These are placeholder imports - adjust based on your actual implementations
try:
    from http_gateway_client import HttpGatewayClient
except ImportError:
    class HttpGatewayClient:
        """Placeholder HttpGatewayClient class."""
        def __init__(self, base_url: str, timeout: int = 30, verify_ssl: bool = True):
            self.base_url = base_url
            self.timeout = timeout
            self.verify_ssl = verify_ssl

        def get_config(self, request_data: Dict[str, Any]) -> Dict[str, Any]:
            """Get config method placeholder."""
            return {}

        def update_log(self, request):
            """Update log method placeholder."""
            pass

        def update_event(self, request):
            """Update event method placeholder."""
            pass


try:
    from common_response import CommonResponse
except ImportError:
    class CommonResponse:
        """Placeholder CommonResponse class."""
        pass


try:
    from update_log_request import UpdateLogRequest
except ImportError:
    class UpdateLogRequest:
        """Placeholder UpdateLogRequest class."""
        pass


try:
    from update_event_request import UpdateEventRequest
except ImportError:
    class UpdateEventRequest:
        """Placeholder UpdateEventRequest class."""
        pass


class DllpgdLiteSDK:
    """
    DllpgdLite SDK - Main SDK class for data logging and analytics.

    This class provides comprehensive methods for:
    - Configuration retrieval
    - Log updates
    - Event tracking
    - Sample data generation for testing

    Attributes:
        http_client: HTTP gateway client instance for making requests
    """

    # Singleton instance
    # Note: Original Java code uses obfuscated base URL
    # You should replace this with your actual URL
    INSTANCE: Optional['DllpgdLiteSDK'] = None

    def __init__(self, base_url: str, timeout: int = 30, verify_ssl: bool = True):
        """
        Initialize DllpgdLiteSDK.

        Args:
            base_url: Base URL for the API endpoint
            timeout: Request timeout in seconds (default: 30)
            verify_ssl: Whether to verify SSL certificates (default: True)
        """
        self.http_client = HttpGatewayClient(base_url, timeout, verify_ssl)

    @classmethod
    def get_instance(cls, base_url: Optional[str] = None) -> 'DllpgdLiteSDK':
        """
        Get or create singleton instance.

        Args:
            base_url: Base URL for the API endpoint (required on first call)

        Returns:
            DllpgdLiteSDK singleton instance
        """
        if cls.INSTANCE is None:
            if base_url is None:
                raise ValueError("base_url is required for first initialization")
            cls.INSTANCE = cls(base_url)
        return cls.INSTANCE

    @staticmethod
    def create_sample_atom() -> Dict[str, Any]:
        """
        Create a sample atom object for testing.

        This method generates a sample data structure representing an "atom"
        which is the basic unit of tracked data in the SDK.

        Returns:
            Dictionary containing sample atom data

        Note:
            The original Java code uses obfuscated string keys.
            This version uses descriptive key names for clarity.
            Adjust the keys based on your actual API requirements.
        """
        # Main atom object
        atom = {
            # Basic identification fields
            "app_id": "com.example.app",  # Decrypted from obfuscated bytes
            "version": 208,
            "platform": "android",
            "device_id": "device_12345",
            "user_id": "user_67890",
            "session_id": "session_abc123",
            "custom_field": "custom_value",
            "enable_feature": False,

            # Nested header object
            "header": {
                "os": "Android",
                "os_version": "11.0",
                "app_version": "1.0.0",
                "sdk_version": "2.0",
                "timestamp": int(time.time() * 1000)  # Current timestamp in milliseconds
            },

            # Events array
            "events": []
        }

        return atom

    @staticmethod
    def create_sample_logs() -> List[Dict[str, Any]]:
        """
        Create sample log entries for testing.

        Returns:
            List of dictionaries containing sample log data

        Note:
            The original Java code uses obfuscated string keys.
            This version uses descriptive key names for clarity.
        """
        logs = []

        log_entry = {
            "level": 1,  # Log level (1=INFO, 2=WARNING, 3=ERROR, etc.)
            "message": "Sample log message",
            "tag": "SampleTag",
            "timestamp": int(time.time() * 1000)
        }

        logs.append(log_entry)
        return logs

    @staticmethod
    def create_sample_events() -> List[Dict[str, Any]]:
        """
        Create sample event entries for testing.

        Returns:
            List of dictionaries containing sample event data

        Note:
            The original Java code uses obfuscated string keys.
            This version uses descriptive key names for clarity.
        """
        events = []

        event_entry = {
            "name": "sample_event",
            "type": "user_action",
            "timestamp": int(time.time() * 1000)
        }

        events.append(event_entry)
        return events

    def get_config(self, request_data: Dict[str, Any]) -> Dict[str, Any]:
        """
        Retrieve configuration from the server.

        Args:
            request_data: Dictionary containing request parameters

        Returns:
            Dictionary containing configuration data from server

        Raises:
            Exception: If the request fails
        """
        return self.http_client.get_config(request_data)

    def update_log(self, update_log_request: UpdateLogRequest) -> CommonResponse:
        """
        Send log update request to the server.

        Args:
            update_log_request: Request object containing log data

        Returns:
            CommonResponse from the server

        Raises:
            Exception: If the request fails
        """
        return self.http_client.update_log(update_log_request)

    def update_event(self, update_event_request: UpdateEventRequest) -> CommonResponse:
        """
        Send event update request to the server.

        Args:
            update_event_request: Request object containing event data

        Returns:
            CommonResponse from the server

        Raises:
            Exception: If the request fails
        """
        return self.http_client.update_event(update_event_request)


# Example usage and test
if __name__ == "__main__":
    import json

    print("=== DllpgdLiteSDK Test ===\n")

    # Test initialization
    base_url = "https://api.example.com"
    sdk = DllpgdLiteSDK(base_url)
    print(f"SDK initialized with base URL: {base_url}\n")

    # Test sample data generation
    print("1. Sample Atom:")
    sample_atom = DllpgdLiteSDK.create_sample_atom()
    print(json.dumps(sample_atom, indent=2))
    print()

    print("2. Sample Logs:")
    sample_logs = DllpgdLiteSDK.create_sample_logs()
    print(json.dumps(sample_logs, indent=2))
    print()

    print("3. Sample Events:")
    sample_events = DllpgdLiteSDK.create_sample_events()
    print(json.dumps(sample_events, indent=2))
    print()

    # Test with custom parameters
    print("4. Custom SDK initialization:")
    sdk_custom = DllpgdLiteSDK(
        base_url="https://api.example.com",
        timeout=60,
        verify_ssl=True
    )
    print(f"Custom SDK initialized with timeout=60, verify_ssl=True\n")

    # Test singleton pattern
    print("5. Singleton pattern test:")
    DllpgdLiteSDK.INSTANCE = None  # Reset for testing
    instance1 = DllpgdLiteSDK.get_instance("https://api.example.com")
    instance2 = DllpgdLiteSDK.get_instance()
    print(f"Same instance: {instance1 is instance2}\n")

    print("=== All tests completed ===")

    # Note: Actual usage would require proper request objects
    # Example:
    # config_request = {"app_id": "com.example.app"}
    # config_response = sdk.get_config(config_request)
    #
    # log_request = UpdateLogRequest(...)
    # log_response = sdk.update_log(log_request)
    #
    # event_request = UpdateEventRequest(...)
    # event_response = sdk.update_event(event_request)
